### ISSUE TITLE
Axis Labels near Zero in Colormap

### ISSUE THREAD

**Issue by jvannugteren** [epezent/implot#176]
First of all thank you very much for sharing this great library. It has been very useful for my project, which involves magnetic field calculation for coils and permanent magnets.

While using the heatmap I found a graphical bug in the color scale that sometimes appears but not always. The axis label near zero sometimes goes to scientific notation, failing to just round to 0.0. I'm using the master branch of your repository. I have attached a screenshot to illustrate the issue.

![image](https://user-images.githubusercontent.com/78733220/107222307-08d58300-6a15-11eb-8840-dd1620a660ac.png)

**Comment by epezent** [comment]
@jvannugteren , would you mind producing a minimal example of this so I can debug the issue?

**Comment by jvannugteren** [comment]
Thanks for having a look. 

I made a minimalistic example case. See code below.

```
const float lim1 = -0.14;
const float lim2 = 0.14;
const float height = 400;
ImGui::SetNextWindowSize(ImVec2(600,500), ImGuiCond_Once);
if(ImGui::Begin("myid")){
    if(ImPlot::BeginPlot("blaat","x-axis","y-axis",{ImGui::GetWindowWidth()-100,height})){
        // it doesn't matter what you plot
        ImPlot::EndPlot();
    }
    ImGui::SameLine();
    ImPlot::ShowColormapScale(lim1, lim2, height);
}
ImGui::End();
```

This gives me the following result:
![image](https://user-images.githubusercontent.com/78733220/109314827-221a6400-784a-11eb-8dd0-c55270bcb135.png)

When lim1 and lim2 (in the example) are for example minus and plus 1.4, the colorscale is shown correctly. Therefore I suspect some sort of rounding error. Or fail to detect zero due to numerical precision.

**Comment by epezent** [comment]
Thanks. I'll give this a look soon!

**Comment by epezent** [comment]
@jvannugteren, can you let me know if this commit https://github.com/epezent/implot/commit/b958abf459e63a039471925eddcd869aeec389c0 resolves your issues?

**Comment by jvannugteren** [comment]
Unfortunately not completely. The example with limits -0.14 to 0.14 is now fine. 

However, the following example still produces the problem:

```
// lock next window position
const float lim1 = -1.24;
const float lim2 = 0.45;
const float height = 400;
ImGui::SetNextWindowSize(ImVec2(600,500), ImGuiCond_Once);
if(ImGui::Begin("myid")){
    if(ImPlot::BeginPlot("blaat","x-axis","y-axis",{ImGui::GetWindowWidth()-100,height})) {
        ImPlot::EndPlot();
    }
    ImGui::SameLine();
    ImPlot::ShowColormapScale(lim1, lim2, height);
}
ImGui::End();
```

![image](https://user-images.githubusercontent.com/78733220/109829333-b5cea480-7c3d-11eb-9a7a-544d87dc5ffd.png)

**Comment by epezent** [comment]
I just pushed another fix that should resolve the issue. In addition, I took another look at this function and added a few improvements. Namely, the signature has change to accept an `ImVec2 size` instead of just the height. Leave the x value at 0 for the previous auto-sizing policy, or set it to a desired value for a fixed width widget. 

https://github.com/epezent/implot/commit/46ea9abab4bb8b5fb60ce6b61f6e5c1bf2e840c1

If this resolves your issue, feel free to close. Thanks!
